docker/daemon/daemon_dest.go

```diff
@@ -22,36 +17,49 @@ import (
 
 type daemonImageDestination struct {
 	ref daemonReference
-	// For talking to imageLoadGoroutine
+	img *dockerArchive.Image
+
+	// Related to the goroutine which runs the `docker load`.
 	goroutineCancel context.CancelFunc
 	statusChannel   <-chan error
-	writer          *io.PipeWriter
-	tar             *tar.Writer
-	// Other state
-	committed bool // writer has been closed
+	committed       bool
+
+	// Horrible hack.
```

Comment:
If this is a hack, it would be nice for the comment to say what the hack is and what prevents it from becoming better. As is, this comment does not really say anything beyond vague “beware” and there is no way to tell when the original reason for the comment no longer applies and the comment can be removed.